### PR TITLE
Fix overzoomed collision debug labels

### LINKED ISSUE #2033: collisionDebug boxes don't show up at zoom levels higher than 16
collisionDebug boxes mysteriously disappear when the map zoom level moves about z16. I'm not sure if this is a bug or intended behavior, but if it's intended it likely needs to be documented.

### PR DESCRIPTION
**Pull request by lucaswoj**
fixes #2033

cc @tmcw @ansis @bhousel

### CODE REVIEW

File: js/render/draw_collision_debug.js [hunk 1]
```diff
@@ -2,33 +2,39 @@
...
+        gl.uniform1f(shader.u_scale, Math.pow(2, painter.transform.zoom - tile.coord.z));
+        gl.uniform1f(shader.u_zoom, painter.transform.zoom * 10);
+        gl.uniform1f(shader.u_maxzoom, (tile.coord.z + 1) * 10);
 
-    gl.uniform1f(shader.u_scale, Math.pow(2, painter.transform.zoom - tile.coord.z));
-    gl.uniform1f(shader.u_zoom, painter.transform.zoom * 10);
-    gl.uniform1f(shader.u_maxzoom, (tile.coord.z + 1) * 10);
+        gl.drawArrays(
+            gl.LINES,
+            elementGroups.groups[0].vertexStartIndex,
```
Comment (ansis) [inline js/render/draw_collision_debug.js]: this line already existed before this pr, but does `.groups[0]` always exist at this point?

File: js/render/draw_collision_debug.js [hunk 2]
```diff
@@ -2,33 +2,39 @@
...
+            elementGroups.groups[0].vertexLength
```
Comment (ansis) [inline js/render/draw_collision_debug.js]: this also existed before this pr, but could this ever be 0 if a layer has no symbol instances? might be causing https://github.com/mapbox/mapbox-gl-js/issues/2032

File: js/render/draw_collision_debug.js (hunk 1 above)
Comment (lucaswoj) [inline js/render/draw_collision_debug.js]: I don't know. It seems like a dangerous assumption. I will fix in a subsequent commit.

File: js/render/draw_collision_debug.js (hunk 2 above)
Comment (lucaswoj) [inline js/render/draw_collision_debug.js]: Investigating this led me to a much worse bug. Check out how we call `makeRoomFor` https://github.com/mapbox/mapbox-gl-js/blob/master/js/data/symbol_bucket.js#L501

**Comment by lucaswoj** [comment]
Just pushed some new commits :eyes: @ansis

**Comment by ansis** [comment]
Since collision box rendering doesn't use indexed rendering (it only has a vertex buffer, no element buffer) there isn't a maximum number of vertices that can be added. The initial `makeRoomFor(8)` (which used to be `makeRoomFor(0)` before fdaf066469ce2f7f213e45febe529d38662707b6) only ensured that the initial element group was created. It was a very unclear, indirect way of doing things.

Maybe we should switch to using indexed rendering for collision boxes. It shouldn't have much of an effect but it would be more consistent with other rendering calls.

**Comment by lucaswoj** [comment]
I understand now. What do you think about these two commits instead?

**Comment by ansis** [comment]
Looks good to me

I'll open a separate issue about switching to indexed rendering so that we can get rid of this confusing exception.

**Pull request merged**